proxy.h

```diff
@@ -871,8 +773,16 @@ class proxy : public details::facade_traits<F>::direct_accessor {
     reset();
     return initialize<P>(il, std::forward<Args>(args)...);
   }
-  auto operator->() const noexcept requires(HasIndirection) { return &ia_; }
-  auto& operator*() const noexcept requires(HasIndirection) { return ia_; }
+  auto operator->() noexcept requires(HasIndirection)
+      { return std::addressof(ia_); }
+  auto operator->() const noexcept requires(HasIndirection)
+      { return std::addressof(ia_); }
+  auto& operator*() & noexcept requires(HasIndirection) { return ia_; }
+  auto& operator*() const& noexcept requires(HasIndirection) { return ia_; }
+  auto&& operator*() && noexcept requires(HasIndirection)
+      { return std::forward<typename Traits::indirect_accessor>(ia_); }
+  auto&& operator*() const&& noexcept requires(HasIndirection)
+      { return std::forward<const typename Traits::indirect_accessor>(ia_); }
```

Comment:
Do we have tests covering these invocations?

Reply:
Good catch! I tried to add a test case and found a bug in the code... Now it's covered.